src/main/resources/META-INF/persistence.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<persistence xmlns="http://java.sun.com/xml/ns/persistence"
+             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+             xsi:schemaLocation="http://java.sun.com/xml/ns/persistence http://java.sun.com/xml/ns/persistence/persistence_2_0.xsd"
+             version="2.0">
+    <persistence-unit name="akka-sample-crud-jpa">
+        <properties>
+            <property name="javax.persistence.jdbc.driver" value="org.hsqldb.jdbcDriver"/>
+            <property name="javax.persistence.jdbc.url" value="jdbc:hsqldb:mem:testdb"/>
+            <property name="javax.persistence.jdbc.user" value="sa"/>
+            <property name="javax.persistence.jdbc.password" value=""/>
+            <!-- field access, no setters -->
+            <property name="hibernate.default-access" value="field"/>
+            <!-- avoid lazy loading -->
```

Comment:
We might want to explain this somewhere. A lot of people using JPA/Hibernate might be used to the lazy loading behavior and be surprised when it doesn't work the way it does in a thread-per-request situation.

Reply:
Adding an example on how to handle optionally-loaded relationships would be useful but it would make the sample a lot more complicated...

(I guess we might want to recommend functions like `<T> T findUserAndConvert(..., Function<User, T> conversion) {}` that perform the conversion inside the session, for those afraid of pushing knowledge of the 'read model' into the repository?)

Reply:
In my opinion a repository should be specific for the use cases needed in the app and not be generic. It can have many different find methods for different needs. Such conversion logic fits well inside the repository.

Reply:
Yes, I agree @patriknw. That's a good pattern. It might be unfamiliar to some of the target audience.

What I really meant with my original comment is that we should explicitly describe somewhere why lazy-loading must be avoided, and how and why entities are detached from the session. It might seem obvious to us, but for people who have never thought about how either JPA or asynchronous execution works, it might not be.

If we can recommend alternative strategies such as use-case specific query methods, even better.